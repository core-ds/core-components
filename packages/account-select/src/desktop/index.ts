export { AccountSelectDesktop } from './Component.desktop';
