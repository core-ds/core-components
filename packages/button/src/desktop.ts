export { ButtonDesktop } from './Component.desktop';
export type { CommonButtonProps as ButtonDesktopProps } from './typings';
