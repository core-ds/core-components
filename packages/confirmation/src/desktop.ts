export * from './component.desktop';

export { useConfirmation, useCountdown } from './utils';
