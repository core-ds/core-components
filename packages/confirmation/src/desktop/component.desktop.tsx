import React, { type FC } from 'react';

import { BaseConfirmation } from '../components/base-confirmation';
import { type ConfirmationProps } from '../types';

export type DesktopConfirmationProps = Omit<ConfirmationProps, 'mobile' | 'breakpoint'>;

export const ConfirmationDesktop: FC<DesktopConfirmationProps> = (props) => (
    <BaseConfirmation {...props} />
);
