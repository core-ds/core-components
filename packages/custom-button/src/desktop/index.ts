export { CustomButtonDesktop } from './Component.desktop';
