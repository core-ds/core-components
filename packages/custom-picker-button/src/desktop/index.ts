export { CustomPickerButtonDesktop, CustomPickerButtonDesktopProps } from '../Component.desktop';
