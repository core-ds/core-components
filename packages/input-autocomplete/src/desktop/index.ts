export { InputAutocompleteDesktop } from '../Component.desktop';
export type { InputAutocompleteCommonProps as InputAutocompleteDesktopProps } from '../types';
