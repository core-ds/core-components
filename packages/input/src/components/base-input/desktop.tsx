import React, { forwardRef } from 'react';

import { ClearButtonDesktop } from '../clear-button';

import { type BaseInputProps, BaseInputComponent } from './Component';

export const BaseInputDesktop = forwardRef<HTMLInputElement, BaseInputProps>(
    (props: BaseInputProps, ref) => (
        <BaseInputComponent {...props} ClearButton={ClearButtonDesktop} ref={ref} />
    ),
);
