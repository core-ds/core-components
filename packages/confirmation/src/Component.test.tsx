import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ConfirmationDesktop, DesktopConfirmationProps } from './desktop';

/**
 * TODO: сделать тесты на все callbacks
 * TODO: сделать тесты на все таймеры
 */
describe('Confirmation', () => {
    const baseProps: DesktopConfirmationProps = {
        screen: 'INITIAL',
        state: 'INITIAL',
        onChangeScreen: jest.fn(),
        onChangeState: jest.fn(),
        onInputFinished: jest.fn(),
        onSmsRetryClick: jest.fn(),
    };

    describe('Snapshot tests', () => {
        it('should match snapshot', () => {
            const { container } = render(<ConfirmationDesktop {...baseProps} />);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with CODE_CHECKING state', () => {
            const { container } = render(
                <ConfirmationDesktop {...baseProps} state='CODE_CHECKING' />,
            );

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with CODE_SENDING state', () => {
            const { container } = render(
                <ConfirmationDesktop {...baseProps} state='CODE_SENDING' />,
            );

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with CODE_ERROR state', () => {
            const { container } = render(<ConfirmationDesktop {...baseProps} state='CODE_ERROR' />);

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with FATAL_ERROR screen', () => {
            const { container } = render(
                <ConfirmationDesktop {...baseProps} screen='FATAL_ERROR' />,
            );

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with HINT screen', () => {
            const { container } = render(
                <ConfirmationDesktop {...baseProps} screen='TEMP_BLOCK' />,
            );

            expect(container).toMatchSnapshot();
        });

        it('should match snapshot with HINT screen', () => {
            const { container } = render(<ConfirmationDesktop {...baseProps} screen='HINT' />);

            expect(container).toMatchSnapshot();
        });
    });

    describe('Props tests', () => {
        it('should render passed texts on INITIAL screen', async () => {
            const texts = {
                title: 'Title',
                linkToHint: 'Link',
                codeError: 'Code error',
                codeChecking: 'Code checking',
                codeSending: 'Code sending',
                hintButton: 'Hint button',
            };

            const props = { ...baseProps, texts };

            const { getByText, rerender } = render(<ConfirmationDesktop {...props} />);

            expect(getByText(texts.title)).toBeInTheDocument();
            expect(getByText(texts.linkToHint)).toBeInTheDocument();

            rerender(<ConfirmationDesktop {...props} state='CODE_CHECKING' />);

            expect(getByText(texts.codeChecking)).toBeInTheDocument();

            rerender(<ConfirmationDesktop {...props} state='CODE_SENDING' />);

            expect(getByText(texts.codeSending)).toBeInTheDocument();

            rerender(<ConfirmationDesktop {...props} state='CODE_ERROR' />);

            expect(getByText(texts.codeError)).toBeInTheDocument();
        });

        it('should render passed texts on HINT screen', () => {
            const texts = {
                hintButton: 'Hint button',
            };

            const props = { ...baseProps, texts };

            const { getByText } = render(<ConfirmationDesktop {...props} screen='HINT' />);

            expect(getByText(texts.hintButton)).toBeInTheDocument();
        });

        it('should render passed texts on FATAL_ERROR screen', () => {
            const texts = {
                fatalErrorTitle: 'Fatal error title',
                fatalErrorDescription: 'Fatal error description',
                fatalErrorButton: 'Fatal error button',
            };

            const props = { ...baseProps, texts };

            const { getByText } = render(<ConfirmationDesktop {...props} screen='FATAL_ERROR' />);

            expect(getByText(texts.fatalErrorTitle)).toBeInTheDocument();
            expect(getByText(texts.fatalErrorDescription)).toBeInTheDocument();
            expect(getByText(texts.fatalErrorButton)).toBeInTheDocument();
        });

        it('should render passed texts on TEMP_BLOCK screen', () => {
            const texts = {
                tempBlockTitle: 'Temp block title',
                tempBlockDescription: 'Temp block description',
            };

            const props = { ...baseProps, texts };

            const { getByText } = render(<ConfirmationDesktop {...props} screen='TEMP_BLOCK' />);

            expect(getByText(texts.tempBlockTitle)).toBeInTheDocument();
            expect(getByText(texts.tempBlockDescription)).toBeInTheDocument();
        });

        it('should set `data-test-id` attribute', () => {
            const testId = 'test-id';
            const { getByTestId } = render(
                <ConfirmationDesktop {...baseProps} dataTestId={testId} />,
            );

            expect(getByTestId(testId)).toBeInTheDocument();
        });

        it('should set custom class', () => {
            const className = 'custom-class';
            const { container } = render(
                <ConfirmationDesktop {...baseProps} className={className} />,
            );

            expect(container.firstElementChild).toHaveClass(className);
        });

        it('should render passed inputs amount', () => {
            const requiredCharAmount = 2;

            const { container } = render(
                <ConfirmationDesktop {...baseProps} requiredCharAmount={requiredCharAmount} />,
            );

            const inputs = container.querySelectorAll('input');

            expect(inputs.length).toBe(requiredCharAmount);
        });

        it('should render passed phone', () => {
            const phone = '+7 (999) 999 99-99';

            const { getByText } = render(<ConfirmationDesktop {...baseProps} phone={phone} />);

            expect(getByText(`Код отправлен на ${phone}`)).toBeInTheDocument();
        });
    });

    it('Should render custom screens', () => {
        const initialScreenTitlle = 'Initial screen title';
        const hintScreenTitlle = 'Hint screen title';
        const fatalErrorScreenTitlle = 'Fatal error screen title';
        const tempBlockScreenTitlle = 'Temp block screen title';

        const props = {
            ...baseProps,
            getScreensMap: (screensMap: any) => ({
                ...screensMap,
                INITIAL: () => <span>{initialScreenTitlle}</span>,
                HINT: () => <span>{hintScreenTitlle}</span>,
                FATAL_ERROR: () => <span>{fatalErrorScreenTitlle}</span>,
                TEMP_BLOCK: () => <span>{tempBlockScreenTitlle}</span>,
            }),
        };

        const { getByText, rerender } = render(<ConfirmationDesktop {...props} />);

        expect(getByText(initialScreenTitlle)).toBeInTheDocument();

        rerender(<ConfirmationDesktop {...props} screen='HINT' />);

        expect(getByText(hintScreenTitlle)).toBeInTheDocument();

        rerender(<ConfirmationDesktop {...props} screen='FATAL_ERROR' />);

        expect(getByText(fatalErrorScreenTitlle)).toBeInTheDocument();

        rerender(<ConfirmationDesktop {...props} screen='TEMP_BLOCK' />);

        expect(getByText(tempBlockScreenTitlle)).toBeInTheDocument();
    });

    describe('Input tests', () => {
        const getActiveElement = () => document.activeElement as Element;

        it('should focus input on first render', () => {
            const { container } = render(<ConfirmationDesktop {...baseProps} />);

            expect(getActiveElement()).toBe(container.querySelector('input'));
        });

        it('should call onInputFinished when input is finished', () => {
            const onInputFinished = jest.fn();

            const { container } = render(
                <ConfirmationDesktop
                    {...baseProps}
                    onInputFinished={onInputFinished}
                    requiredCharAmount={2}
                />,
            );

            const inputs = container.querySelectorAll('input');

            fireEvent.change(inputs[0], { target: { value: '1' } });
            fireEvent.change(inputs[1], { target: { value: '2' } });

            expect(onInputFinished).toBeCalledTimes(1);
        });

        it('should call onFatalErrorOkButtonClick when click on button', () => {
            const onFatalErrorOkButtonClick = jest.fn();

            const { container } = render(
                <ConfirmationDesktop
                    {...baseProps}
                    onFatalErrorOkButtonClick={onFatalErrorOkButtonClick}
                    screen='FATAL_ERROR'
                />,
            );

            const button = container.querySelector('button') as HTMLButtonElement;

            fireEvent.click(button);

            expect(onFatalErrorOkButtonClick).toBeCalledTimes(1);
        });

        it('should call onChangeState when error occurred', async () => {
            const onChangeState = jest.fn();

            const { container } = render(
                <ConfirmationDesktop
                    {...baseProps}
                    state='CODE_ERROR'
                    clearCodeOnError={true}
                    onChangeState={onChangeState}
                />,
            );

            const codeInput = container.querySelector('div[class*="codeInput"]') as HTMLDivElement;

            fireEvent.animationEnd(codeInput);

            await waitFor(() => {
                expect(onChangeState).toBeCalledTimes(1);
                expect(onChangeState).toBeCalledWith('INITIAL');
            });
        });
    });
});
