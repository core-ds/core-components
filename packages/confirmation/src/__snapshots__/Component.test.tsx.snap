// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation Snapshot tests should match snapshot 1`] = `
<div>
  <div
    class="component left"
  >
    <div
      class="component left"
    >
      <h3
        class="component header typography styrene-medium medium primary"
      >
        Введите код из сообщения
      </h3>
      <span
        class="phone typographyTheme primary primary-medium"
      >
        Код отправлен на 
      </span>
      <div
        class="component containerInput codeInput"
      >
        <div
          class=""
        >
          <input
            autocomplete="one-time-code"
            class="input"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
        </div>
      </div>
      <div
        class="countdown countdownContainer typographyTheme primary-medium"
      >
        Запросить повторно можно через
         
        01:00
      </div>
      <button
        class="component secondary pseudo smsComeLink"
        type="button"
      >
        <span
          class="text"
        >
          Не приходит сообщение?
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirmation Snapshot tests should match snapshot with CODE_CHECKING state 1`] = `
<div>
  <div
    class="component left"
  >
    <div
      class="component left"
    >
      <h3
        class="component header typography styrene-medium medium primary"
      >
        Введите код из сообщения
      </h3>
      <span
        class="phone typographyTheme primary primary-medium"
      >
        Код отправлен на 
      </span>
      <div
        class="component containerInput codeInput"
      >
        <div
          class=""
        >
          <input
            autocomplete="one-time-code"
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
        </div>
      </div>
      <div
        class="loaderWrap countdownContainer typographyTheme"
      >
        <div
          class="component"
        >
          <div />
          <div />
          <div />
        </div>
        <span
          class="loaderText"
        />
      </div>
      <button
        class="component secondary pseudo smsComeLink"
        type="button"
      >
        <span
          class="text"
        >
          Не приходит сообщение?
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirmation Snapshot tests should match snapshot with CODE_ERROR state 1`] = `
<div>
  <div
    class="component left"
  >
    <div
      class="component left"
    >
      <h3
        class="component header typography styrene-medium medium primary"
      >
        Введите код из сообщения
      </h3>
      <span
        class="phone typographyTheme primary primary-medium"
      >
        Код отправлен на 
      </span>
      <div
        class="component containerInput codeInput"
      >
        <div
          class="shake"
        >
          <input
            autocomplete="one-time-code"
            class="input hasError"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input hasError"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input hasError"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input hasError"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input hasError"
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
        </div>
        <div
          class="error"
          role="alert"
        >
          Код введён неверно
        </div>
      </div>
      <div
        class="countdown countdownContainer typographyTheme primary-medium"
      >
        Запросить повторно можно через
         
        01:00
      </div>
      <button
        class="component secondary pseudo smsComeLink"
        type="button"
      >
        <span
          class="text"
        >
          Не приходит сообщение?
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirmation Snapshot tests should match snapshot with CODE_SENDING state 1`] = `
<div>
  <div
    class="component left"
  >
    <div
      class="component left"
    >
      <h3
        class="component header typography styrene-medium medium primary"
      >
        Введите код из сообщения
      </h3>
      <span
        class="phone typographyTheme primary primary-medium"
      >
        Код отправлен на 
      </span>
      <div
        class="component containerInput codeInput"
      >
        <div
          class=""
        >
          <input
            autocomplete="one-time-code"
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
          <input
            autocomplete=""
            class="input disabled"
            disabled=""
            inputmode="numeric"
            pattern="[0-9]*"
            value=""
          />
        </div>
      </div>
      <div
        class="loaderWrap countdownContainer typographyTheme"
      >
        <div
          class="component"
        >
          <div />
          <div />
          <div />
        </div>
        <span
          class="loaderText"
        />
      </div>
      <button
        class="component secondary pseudo smsComeLink"
        type="button"
      >
        <span
          class="text"
        >
          Не приходит сообщение?
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirmation Snapshot tests should match snapshot with FATAL_ERROR screen 1`] = `
<div>
  <div
    class="component left"
  >
    <div
      class="component left"
    >
      <h3
        class="component header typography styrene-medium medium primary"
      >
        Ввести код больше нельзя
      </h3>
      <span
        class="typographyTheme primary primary-medium"
      >
        Вы ввели код неверно более 5 раз. В целях безопасности мы ограничили дальнейший ввод.
      </span>
      <button
        class="component secondary s component secondary button"
        type="button"
      >
        <span
          class="text stretchText"
        >
          Понятно
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Confirmation Snapshot tests should match snapshot with HINT screen 1`] = `
<div>
  <div
    class="component left"
  >
    <h3
      class="component header typography styrene-medium medium primary"
    >
      Превышено количество попыток запроса кода
    </h3>
    <span
      class="description typographyTheme primary primary-medium"
    >
      Повторное подтверждение кодом будет возможно через 24 часа
    </span>
    <div
      class="countdownWrap typographyTheme"
    >
      <svg
        class="loader"
        height="16"
        viewBox="0 0 16 16"
        width="16"
      >
        <circle
          class="circle"
          cx="8"
          cy="8"
          r="7"
          stroke-dasharray="43.98"
          stroke-dashoffset="0.00"
          transform="rotate(-90 8 8)"
        />
      </svg>
      24:00:00
    </div>
  </div>
</div>
`;

exports[`Confirmation Snapshot tests should match snapshot with HINT screen 2`] = `
<div>
  <div
    class="component left"
  >
    <div
      class="component left"
    >
      <div
        class="hintWrap"
      >
        <h3
          class="component header typography styrene-medium medium primary"
        >
          Не приходит сообщение?
        </h3>
        <span
          class="text typographyTheme primary primary-medium"
        >
          Если у вас изменился номер телефона, позвоните нам или обратитесь в любое отделение банка.
        </span>
        <div
          class="phonesWrap"
        >
          <div
            class="phoneWrap"
          >
            <a
              class="component primary withoutUnderline phoneLink"
              href="tel:+78002000000"
            >
              <span
                class="text"
              >
                8 800 200-00-00
              </span>
            </a>
            <span
              class="text typographyTheme primary primary-medium"
            >
               — для звонков по России
            </span>
          </div>
          <div
            class="phoneWrap"
          >
            <a
              class="component primary withoutUnderline phoneLink"
              href="tel:+74957888878"
            >
              <span
                class="text"
              >
                +7 495 78-888-78
              </span>
            </a>
            <span
              class="text typographyTheme primary primary-medium"
            >
               — в Москве и за границей
            </span>
          </div>
        </div>
        <span
          class="text typographyTheme primary primary-medium"
        >
          Если номер не менялся, возможно, перегружен сервис отправки сообщений. Попробуйте повторить действие через несколько минут.
        </span>
        <button
          class="component secondary s component secondary hintButton"
          type="button"
        >
          <span
            class="text stretchText"
          >
            Вернуться к вводу кода
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
